packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -202,48 +128,53 @@ bool _isJsonMapStyle(Map value) {
 }
 
 // Converts an incoming JSON-encoded Style info, into the correct gmaps array.
-List<gmaps.MapTypeStyle> _mapStyles(String mapStyleJson) {
+List<gmaps.MapTypeStyle?> _mapStyles(String? mapStyleJson) {
   List<gmaps.MapTypeStyle> styles = [];
   if (mapStyleJson != null) {
-    styles = json.decode(mapStyleJson, reviver: (key, value) {
-      if (value is Map && _isJsonMapStyle(value)) {
-        return gmaps.MapTypeStyle()
-          ..elementType = _elementTypeToEnum[value['elementType']]
-          ..featureType = _featureTypeToEnum[value['featureType']]
-          ..stylers = _parseStylers(value['stylers']);
-      }
-      return value;
-    }).cast<gmaps.MapTypeStyle>();
+    styles = json
+        .decode(mapStyleJson, reviver: (key, value) {
+          if (value is Map && _isJsonMapStyle(value)) {
+            return gmaps.MapTypeStyle()
+              ..elementType = value['elementType']
+              ..featureType = value['featureType']
+              ..stylers =
+                  (value['stylers'] as List).map((e) => jsify(e)).toList();
+          }
+          return value;
+        })
+        .cast<gmaps.MapTypeStyle>()
+        .toList();
+    // .toList calls are required so the JS API understands the underlying data structure.
   }
   return styles;
 }
 
-gmaps.LatLng _latLngToGmLatLng(LatLng latLng) {
+gmaps.LatLng? _latLngToGmLatLng(LatLng latLng) {
```

Comment:
It looks like this can only be null if the param is null, which would already be a programming error. It seems like we shouldn't bake this into the API just for mixed mode support.

Reply:
Ohh, good catch on this file. Let me give it another pass. Thanks for noticing!

Reply:
I don't understand why the next line of code didn't fail :/

Reply:
>I don't understand why the next line of code didn't fail :/

Because of 
https://github.com/flutter/plugins/blob/15d4f9c4e06014225807c5fbcf681bd6ea1f765d/analysis_options.yaml#L49-L50 ?

Reply:
(x_x) totally